target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-eap1300.dtsi

```diff
@@ -0,0 +1,237 @@
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power: orange {
+			label = "orange:power";
+			gpios = <&tlmm 58 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "blue:lan";
+			gpios = <&tlmm 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		mesh {
+			label = "blue:mesh";
+			gpios = <&tlmm 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan2g {
+			label = "blue:wlan2g";
+			gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g {
+			label = "yellow:wlan5g";
+			gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&tlmm {
+	serial_pins: serial_pinmux {
+		mux {
+			pins = "gpio60", "gpio61";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+
+	spi_0_pins: spi_0_pinmux {
+		pin {
+			function = "blsp_spi0";
+			pins = "gpio54", "gpio55", "gpio56", "gpio57";
+			drive-strength = <12>;
+			bias-disable;
+		};
+		pin_cs {
+			function = "gpio";
+			pins = "gpio54";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>;
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <24000000>;
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:SBL1";
+				reg = <0x00000000 0x00040000>;
+				read-only;
+			};
+			partition1@40000 {
+				label = "0:MIBIB";
+				reg = <0x00040000 0x00020000>;
+				read-only;
+			};
+			partition2@60000 {
+				label = "0:QSEE";
+				reg = <0x00060000 0x00060000>;
+				read-only;
+			};
+			partition3@c0000 {
+				label = "0:CDT";
+				reg = <0x000c0000 0x00010000>;
+				read-only;
+			};
+			partition4@d0000 {
+				label = "0:DDRPARAMS";
+				reg = <0x000d0000 0x00010000>;
+				read-only;
+			};
+			partition5@e0000 {
+				compatible = "u-boot,env";
+				label = "0:APPSBLENV";
+				reg = <0x000e0000 0x00010000>;
+
+				macaddr_lan: ethaddr {
+				};
+			};
+			partition6@f0000 {
+				label = "0:APPSBL";
+				reg = <0x000f0000 0x00090000>;
+				read-only;
+			};
+			partition7@180000 {
+				label = "0:ART";
+				reg = <0x00180000 0x00010000>;
+				read-only;
+				compatible = "nvmem-cells";
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				precal_art_1000: precal@1000 {
+					reg = <0x1000 0x2f20>;
+				};
+
+				precal_art_5000: precal@5000 {
+					reg = <0x5000 0x2f20>;
+				};
+			};
+			partition8@190000 {
+				compatible = "denx,fit";
+				label = "firmware";
+				reg = <0x190000 0x1dc0000>;
+			};
+			partition9@1f50000 {
+				/* seemingly not read by u-boot!? */
```

Comment:
What do you mean by that? Did you run `saveenv` in Uboot while connected via serial to see whether it saves to this partition?
Is is helpful to add this here?

Reply:
I did printenv while connected to serial, then compared the data in the two partitions to confirm which was used. The vendor firmware has the same setup, I'm not sure why.

I haven't "added" anything, other than the comment - see `qcom-ipq4018-eap1300.dts` in `master`.

Reply:
Which two partitions? Is `partition5` the actual uboot environment printenv reads from?

Reply:
I am loath to mess with the data in the `u-boot-env` partition unless I am able to understand what it actually does, and I don't particularly feel like reversing the vendor modified u-boot.

Reply:
then just leave out the comment, if you aren't sure ^^
> then compared the data in the two partitions

which **two** partitions did you mean?

Reply:
> then just leave out the comment, if you aren't sure ^^

I'm sure u-boot doesn't load the environment from it. I don't know if it does something else with it. It might be best to rename the partition in addition to putting a comment there to ensure there's no confusion, but I'm not sure what the best way to explain the situation is.

> > then compared the data in the two partitions
> 
> which **two** partitions did you mean?

`0:APPSBLENV` and `u-boot-env`.